graphics/palette.cpp

```diff
@@ -94,4 +94,19 @@ byte PaletteLookup::findBestColor(byte cr, byte cg, byte cb, bool useNaiveAlg) {
 	return bestColor;
 }
 
+uint32 *PaletteLookup::createMap(const byte *srcPalette, uint len, bool useNaiveAlg) {
+	if (len <= _paletteSize && memcmp(_palette, srcPalette, len) == 0)
+		return nullptr;
+
+	uint32 *map = new uint32[len];
+	for (uint i = 0; i < len; i++) {
+		byte r = *srcPalette++;
+		byte g = *srcPalette++;
+		byte b = *srcPalette++;
+
+		map[i] = findBestColor(r, g, b, useNaiveAlg);
+	}
+	return map;
+}
```

Comment:
@coderabbitai can you check the consideration points for this review to see if there is something we ask you not use in this code review?

Reply:
@coderabbitai Please consider the following project guidelines points:
* We do not use C++ exceptions
* We do not use the std C++ library in our codebase, therefore, we do not use STL classes in our codebase
*  We have predefined common integer types. Use them instead of rolling your own: byte, int8, uint8, int16, uint16, int32, uint32
* All the input/output is performed using our own code which is abstracted in backends directory

Then rewrite or discard your current review.